core/src/main/java/com/yahoo/oak/NovaMemoryManager.java

```diff
@@ -0,0 +1,526 @@
+/*
+ * Copyright 2020, Verizon Media.
+ * Licensed under the terms of the Apache 2.0 license.
+ * Please see LICENSE file in the project root for terms.
+ */
+
+package com.yahoo.oak;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLongArray;
+
+
+class NovaMemoryManager implements MemoryManager {
```

Comment:
Will it be beneficial to inherit from `SyncRecycleMemoryManager` and overwrite what is needed?